lib/state_machines/machine.rb

```diff
@@ -658,6 +658,12 @@ def dynamic_initial_state?
     #   directly to the object
     def initialize_state(object, options = {})
       state = initial_state(object)
+
+      unless object.respond_to? attribute
+        object.class.send(:alias_method, attribute, :state)
```

Comment:
sounds good, btw this was the [failing tests](https://github.com/state-machines/state_machines-activerecord/blob/9b1f27a264560f7cdef91cae56d3aae2ffa30943/test/machine_with_non_column_state_attribute_undefined_test.rb) related on sm-ar. I'll ammend it cheers